packages/kokkos-kernels/blas/src/KokkosBlas1_axpby.hpp

```diff
@@ -17,164 +17,312 @@
+///                        stored.
 template <class execution_space, class AV, class XMV, class BV, class YMV>
-void axpby(const execution_space& space, const AV& a, const XMV& X, const BV& b,
-           const YMV& Y) {
-  static_assert(Kokkos::is_execution_space_v<execution_space>,
-                "KokkosBlas::axpby: execution_space must be a valid Kokkos "
-                "execution space.");
-  static_assert(Kokkos::is_view<XMV>::value,
-                "KokkosBlas::axpby: "
-                "X is not a Kokkos::View.");
-  static_assert(
-      Kokkos::SpaceAccessibility<execution_space,
-                                 typename XMV::memory_space>::accessible,
-      "KokkosBlas::axpby: XMV must be accessible from execution_space");
-  static_assert(Kokkos::is_view<YMV>::value,
-                "KokkosBlas::axpby: "
-                "Y is not a Kokkos::View.");
-  static_assert(
-      Kokkos::SpaceAccessibility<execution_space,
-                                 typename YMV::memory_space>::accessible,
-      "KokkosBlas::axpby: XMV must be accessible from execution_space");
-  static_assert(std::is_same<typename YMV::value_type,
-                             typename YMV::non_const_value_type>::value,
-                "KokkosBlas::axpby: Y is const.  It must be nonconst, "
-                "because it is an output argument "
-                "(we must be able to write to its entries).");
-  static_assert(int(YMV::rank) == int(XMV::rank),
-                "KokkosBlas::axpby: "
-                "X and Y must have the same rank.");
-  static_assert(YMV::rank == 1 || YMV::rank == 2,
-                "KokkosBlas::axpby: "
-                "XMV and YMV must either have rank 1 or rank 2.");
-
-  // Check compatibility of dimensions at run time.
-  if (X.extent(0) != Y.extent(0) || X.extent(1) != Y.extent(1)) {
-    std::ostringstream os;
-    os << "KokkosBlas::axpby: Dimensions of X and Y do not match: "
-       << "X: " << X.extent(0) << " x " << X.extent(1) << ", Y: " << Y.extent(0)
-       << " x " << Y.extent(1);
-    KokkosKernels::Impl::throw_runtime_exception(os.str());
-  }
+void axpby(const execution_space& exec_space, const AV& a, const XMV& X,
+           const BV& b, const YMV& Y) {
+  using AxpbyTraits =
+      Impl::AxpbyUnificationAttemptTraits<execution_space, AV, XMV, BV, YMV>;
+  using InternalTypeA = typename AxpbyTraits::InternalTypeA;
+  using InternalTypeX = typename AxpbyTraits::InternalTypeX;
+  using InternalTypeB = typename AxpbyTraits::InternalTypeB;
+  using InternalTypeY = typename AxpbyTraits::InternalTypeY;
+
+  // **********************************************************************
+  // Perform compile time checks and run time checks.
+  // **********************************************************************
+  AxpbyTraits::performChecks(a, X, b, Y);
+#ifdef HAVE_KOKKOSKERNELS_DEBUG
+  AxpbyTraits::printInformation(std::cout, "axpby(), unif information");
```

Comment:
This line is triggering the MueLu test failures in the host builds; commenting it out allowed tests to pass MueLu_ParameterListInterpreterTpetra and MueLu_CreateOperatorTpetra to pass in the gcc/8.3.0 serial debug build